crates/autopilot/src/run_loop.rs

```diff
@@ -525,6 +532,35 @@ impl RunLoop {
         solutions
     }
 
+    /// Chooses the winners from the given participants.
+    ///
+    /// Participants are already sorted by their score (worst to best).
+    ///
+    /// Winners are selected one by one, starting from the best solution,
+    /// until `max_winners_per_auction` is hit. The solution can become winner
+    /// it is swaps tokens that are not yet swapped by any other already
+    /// selected winner.
+    fn select_winners(&self, participants: &[Participant]) -> Vec<Participant> {
+        let mut winners = Vec::new();
+        let mut already_swapped_tokens = HashSet::new();
+        for participant in participants.iter().rev() {
+            let swapped_tokens = participant
+                .solution
+                .orders()
+                .iter()
+                .map(|(_, order)| (order.sell.token, order.buy.token))
+                .collect::<HashSet<_>>();
+            if swapped_tokens.is_disjoint(&already_swapped_tokens) {
```

Comment:
So, we discard the whole solution if a single order from it contains already pending to swap token. But is it possible to only filter out those orders and sending the remaining ones to the solver via the /settle request?

Reply:
This was not proposed to do by solver team. I don't think it's smart to alter the solution already prepared by solvers as we can't be sure how it affects it's execution. Also, any initiative in the future, based on a hashed commitment on a solver solution would be impossible.

Also, by removing an order from solution it's score changes so we would probably have to rescore the solution. Also, the driver API /settle call would have to be changed to specify which orders are allowed to be settled etc. Overall seems too complicated to bother with atm.